source/proof_resolution/lib/proof_resolution.rb

```diff
@@ -9,38 +9,48 @@
 module IdentityIdpFunctions
   class ProofResolution
     include IdentityIdpFunctions::FaradayHelper
+    include IdentityIdpFunctions::LoggingHelper
 
     def self.handle(event:, context:, &callback_block) # rubocop:disable Lint/UnusedMethodArgument
       params = JSON.parse(event.to_json, symbolize_names: true)
       new(**params).proof(&callback_block)
     end
 
-    attr_reader :applicant_pii, :callback_url, :should_proof_state_id
+    attr_reader :applicant_pii, :callback_url, :should_proof_state_id, :trace_id, :timer
 
-    def initialize(applicant_pii:, callback_url:, should_proof_state_id:)
+    def initialize(applicant_pii:, callback_url:, should_proof_state_id:, trace_id: nil)
```

Comment:
trace_id will always come from the IDP?

Reply:
That was my plan. The IDP will receive X-Amz-Trace-ID HTTP headers, so to keep things part of the same trace, we take that value and pass it here